csrc/dispatch.h

```diff
@@ -165,13 +168,18 @@ class TORCH_CUDA_CU_API OptOutConstDispatch : public PolymorphicBase {
 
   virtual void handle(const AggregateVal*);
 
+  virtual void handleUntypedVal(const Val*);
```

Comment:
I think we should clean this up by making `Scalar` not a template class, and uses `std::any` to store its value. But I will not work on it in this PR.

Reply:
Oh, no, `std::any` do not have `operator==`....

Reply:
Since there are only a handful of scalar types, could we use `std::variant` instead of `std::any`?

Reply:
> Since there are only a handful of scalar types, could we use `std::variant` instead of `std::any`?

No, we can have unlimited number of types:
```
Array<float, 3>
Array<float, 4>
Array<Array<float, 3>, 4>
...
```